pkg/utils/sqlite/sqlite.go

```diff
@@ -0,0 +1,123 @@
+/*
+Package sqlite implements a connect hook around the sqlite3 driver so that the
+underlying connection can be fetched from the driver for more advanced operations such
+as backups. See: https://github.com/mattn/go-sqlite3/blob/master/_example/hook/hook.go
+
+To use make sure you import this package so that the init code registers the driver:
+
+	import _ github.com/rotationalio/ensign/pkg/utils/sqlite
+
+Then you can use sql.Open in the same way you would with sqlite3:
+
+	sql.Open("ensign_sqlite3", "path/to/database.db")
+*/
+package sqlite
```

Comment:
Dang that is kind of a mess, if we did fetch the wrong connection would it corrupt the backup process or trigger an error?

Reply:
It would likely trigger a panic … at least that's what I'm hoping. I think the most likely thing is that the src connection is returned and I'm not sure what happens when you try to backup the same database into itself. 

So yeh … a real mess; but that's why I did the long running test.